internal/gengapic/genrest.go

```diff
@@ -406,6 +423,21 @@ func (g *generator) generateQueryString(m *descriptor.MethodDescriptorProto) {
 		g.imports[pbinfo.ImportSpec{Path: "fmt"}] = true
 		paramAdd := fmt.Sprintf("params.Add(%q, fmt.Sprintf(%q, req%s))", lowerFirst(snakeToCamel(path)), "%v", accessor)
 
+		// Handle well known protobuf types with special JSON encodings.
+		if strContains(wellKnownTypes, field.GetTypeName()) {
+			b := strings.Builder{}
+			b.WriteString(fmt.Sprintf("%s, err := protojson.Marshal(req%s)\n", field.GetJsonName(), accessor))
+			b.WriteString("if err != nil {\n")
+			if m.GetOutputType() == emptyType {
+				b.WriteString(fmt.Sprintf("  return err\n"))
+			} else {
+				b.WriteString(fmt.Sprintf("  return nil, err\n"))
+			}
+			b.WriteString("}\n")
+			b.WriteString(fmt.Sprintf("params.Add(%q, string(%s))", lowerFirst(snakeToCamel(path)), field.GetJsonName()))
+			paramAdd = b.String()
```

Comment:
So you're overwriting the `parmAdd` from above, line 424? What about  registering the `fmt` import in line 423? Is that still valid? I know Go style tries to avoid `else`s, but given the non-trivial processing in both branches I think it might be clearer to organize this as
```
if wellKnownType {
…
} else {
…
}
```

Reply:
Good point about the `fmt` import, I need to conditionally include that. 

I'm not sure I understand the scope of your "if/else" comment though. Do you just mean WRT when to import `fmt`? If so, yeah that is what I plan to do :)

Reply:
PTAL :)

Reply:
Yeah, that's exactly what I meant. Thanks!